src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -144,10 +145,8 @@ public SnowflakeSinkServiceV2(
             Utils.isIcebergEnabled(connectorConfig), schematizationEnabled);
     this.icebergTableSchemaValidator = new IcebergTableSchemaValidator(conn);
     this.icebergInitService = new IcebergInitService(conn);
-    this.schemaEvolutionService =
-        Utils.isIcebergEnabled(connectorConfig)
-            ? new IcebergSchemaEvolutionService(conn)
-            : new SnowflakeSchemaEvolutionService(conn);
+    this.schemaEvolutionService = new SnowflakeSchemaEvolutionService(conn);
```

Comment:
I liked the idea of strategy pattern for `IcebergSchemaEvolutionService` and `SnowflakeSchemaEvolutionService` because we could hide some context from 'DirectTopicPartitionChannel`. Can we just pass `channel.getTableSchema()` even when it's not used for non-iceberg?

Reply:
Having method with unused field is a drawback hence I tried to go the new way, I thought it'a a lesser evil. 
Anyway I can go back to previous state.